### PR TITLE
[Anduril] Defend for containers (cloud-defend) spec updates

### PR DESCRIPTION
**Pull request by mitodrummer**
## What does this PR do?

-Removes cloud-defend from the list of supported metrics components (we aren't using libbeat, and are seeing many errors in the agent logs because of this)
-Adds a spec runtime to check to require root access for cloud-defend binary
-Adds linux to platforms in cloud-defend.spec.yml
-Adds volume/volumeMounts that cloud-defend service requires

## Why is it important?

To allow cloud-defend to be deployed seamlessly and with minimal extra steps. 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
- [x] I have made corresponding change to the default configuration files
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~

### CODE REVIEW

File: specs/cloud-defend.spec.yml
Comment (mitodrummer) [inline specs/cloud-defend.spec.yml:13]: @rhysre Looks like you can check other criteria other than root access. see [specs/endpoint-security.spec.yml](https://github.com/elastic/elastic-agent/blob/main/specs/endpoint-security.spec.yml#L17)

not sure if this is needed, but thought i'd check with you.

**Comment by cmacknz** [comment]
The Go and spec file changes here LGTM. Holding off on approving until @ChrsMark or @MichaelKatsoulis approve the K8S yaml changes.

**Review by ChrsMark — APPROVED** [review]
Changes in the manifests look good to me! However I have a concern about the amount of extra mounts we are adding with the time being. Despite this being what we have agreed on doing so far (to add all the required mounts to improve all user experiences), I have the concern if all these additions would be valid in environments like Openshift or GKE Autopilot. Recently we found out that we need to remove a bunch of mounts in order for our manifests to work on GKE Autopilot. @gizas what do you think here? Is this bearable so far or we need to figure out a more efficient way of managing all these extra requirements per specific use case?

**Comment by MichaelKatsoulis** [comment]
The problem with changes like these in the Kubernetes Manifests is that it gets the onboarding experience of a user in Kubernetes observability way poorer. 
This Manifest will land in Kibana/Fleet-UI agent installation process were the user downloads it and applies it.
In the case that they have selected Kubernetes integration only, they will get a manifest with unnecessary mounts of host path type and it might not even be allowed by their company's security (paths like /sys).
Adding a comment like `remove those lines in case you don't use that integration` adds extra manual steps and can raise the question to the user of `why did you add them in the first place`.
We need to find another way to more cleverly provide the correct manifest to the user depending on what integration they have in their policy. 
This can be done with the use of Kustomize and multiple overlays depending on the need or some more clever code in Kibana.
One problem I foresee is that if a user applies the k8s manifest with only defaults mounts needed for kubernetes integration, then in case they want to add the Defend integration to their policy, agent will break due to missing mounts.

cc @elastic/obs-cloudnative-monitoring  @ruflin

**Comment by ChrsMark** [comment]
> This definitely crossed my mind as I was adding these in. Perhaps it would be better to not add the mounts here, and include these mounts as part of our README doc for the integration itself. At least until it's possible to dynamically provide volumeMounts on a per integration basis (if that is something on the radar). cc @norrietaylor @learhy if you had thoughts on this.
> 
> Our service also requires the agent be run in "privileged: true" mode, so the user would still need to make that change to the yaml in order for the service to spin up correctly.

If you ask from the users to add additional stuff already then it should be better if you add those as commented out sections and indicate to your users to uncomment those to use your features.

**Review by ChrsMark — APPROVED** [review]

**Review by MichaelKatsoulis — APPROVED** [review]

**Comment by mitodrummer** [comment]
Doesn't look like I'm authorized to merge this, even with approvals. Anyone want to do the honors?

**Comment by norrietaylor** [comment]
I would prefer we add the mount points and flags to the YAML upfront. This model has been adapted for other integrations, and consistency is essential. If we have many integrations using this model and one that deviates, it will be confusing for the user.  We should test in all supported environments to ensure these changes do not break additional K8s platforms. 

I am very optimistic about the discussions around declarative automation. I see this as a path forward that would solve this issue. 

@MichaelKatsoulis, are there any plans to support Kustomize further upstream? 

@ChrsMark, does the current YAML work in Autopilot? I was under the impression that Autopilot prevents changes to the `kube-system` namespace. (I could be completely out to lunch here - please correct me if I am wrong ) 

@mitodrummer we should hold off on the merge here until 8.7 is cut as this is planned for 8.8

**Comment by ChrsMark** [comment]
> I would prefer we add the mount points and flags to the YAML upfront. This model has been adapted for other integrations, and consistency is essential. If we have many integrations using this model and one that deviates, it will be confusing for the user. We should test in all supported environments to ensure these changes do not break additional K8s platforms.
> 
> I am very optimistic about the discussions around declarative automation. I see this as a path forward that would solve this issue.
> 

+1 on this!

> @MichaelKatsoulis, are there any plans to support Kustomize further upstream?
> 
> @ChrsMark, does the current YAML work in Autopilot? I was under the impression that Autopilot prevents changes to the `kube-system` namespace. (I could be completely out to lunch here - please correct me if I am wrong )

At the moment no. System mounts and namespace need to be changed so as Agent to run successfully in Autopilot. So we end up with a "reduced" manifest anyways for Autopilot. 
To be honest I have mixed feelings about adding more and more mounts/roles/permissions on the base manifest. However I don't think that we can solve this in that PR so I wouldn't have a strong opinion here. My original comment was mostly to re-raise the concern of overloading the base manifests and that we need to deal with it really soon in a proper and future proof way.

Note: for autopilot we did an extensive research with @gizas to see what features are disabled by reducing the manifest. This is sth extra that we need to consider and define the ownership of such validations for every new set of mounts/roles/permissions that are added in the future.

**Comment by MichaelKatsoulis** [comment]
> @MichaelKatsoulis, are there any plans to support Kustomize further upstream? 

There have been some initial discussions in @elastic/obs-cloudnative-monitoring team of how we can leverage kustomize and multiple overlays(per case) to offer an effortless  onboarding for users in Fleet-UI.
@gizas has created this [issue](https://github.com/elastic/elastic-agent/issues/2175) to gather the steps and needs to be prioritised.

**Pull request merged**
